src/import/converters.rs

```diff
@@ -0,0 +1,775 @@
+                    frame: start_time,
+                    in_tangent: keyframe.base.in_tangent.as_ref().map(conv_keyframe_handle),
+                    out_tangent: keyframe.base.out_tangent.as_ref().map(conv_keyframe_handle),
+                    hold,
+                });
+                values.push(data);
+                // todo: end_value deprecated but should we still push it if it
+                // exists?
+            }
+            Value::Animated(model::Animated {
+                times: frames,
+                values,
+            })
+        }
+    }
+}
+
+pub fn conv_multi<T: Tweenable>(
+    multidimensional: &parser::schema::animated_properties::multi_dimensional::MultiDimensional,
+    f: impl Fn(&Vec<f64>) -> T,
+) -> Value<T> {
+    use crate::parser::schema::animated_properties::animated_property::AnimatedPropertyK::*;
+
+    match &multidimensional.animated_property.value {
+        Static(components) => {
+            let value: Vec<f64> = components
+                .iter()
+                .map(|number| number.as_f64().unwrap())
+                .collect();
+            Value::Fixed(f(&value))
+        }
+        AnimatedValue(keyframes) => {
+            let animated = conv_keyframes(keyframes.iter(), |k| {
+                let data = k
+                    .value
+                    .iter()
+                    .map(|number| number.as_f64().unwrap())
+                    .collect::<Vec<_>>();
+                f(&data)
+            });
+            animated
+        }
+    }
+}
+
+pub fn conv_multi_color<T: Tweenable>(
+    color: &parser::schema::animated_properties::color_value::ColorValue,
+    f: impl Fn(&Vec<f64>) -> T,
+) -> Value<T> {
+    use crate::parser::schema::animated_properties::animated_property::AnimatedPropertyK::*;
+
+    match &color.animated_property.value {
+        Static(components) => {
+            let value: Vec<f64> = components
+                .iter()
+                .map(|number| number.as_f64().unwrap())
+                .collect();
+            Value::Fixed(f(&value))
+        }
+        AnimatedValue(keyframes) => {
+            let animated = conv_keyframes(keyframes.iter(), |k| {
+                let data = k
+                    .value
+                    .iter()
+                    .map(|number| number.as_f64().unwrap())
+                    .collect::<Vec<_>>();
+                f(&data)
+            });
+            animated
+        }
+    }
+}
+
+pub fn conv_pos<T: Tweenable>(
+    position: &parser::schema::animated_properties::position::Position,
+    f: impl Fn(&Vec<f64>) -> T,
+) -> Value<T> {
+    use crate::parser::schema::animated_properties::position::PositionValueK::*;
+
+    match &position.value {
+        Static(components) => {
+            let value: Vec<f64> = components
+                .iter()
+                .map(|number| number.as_f64().unwrap())
+                .collect();
+            Value::Fixed(f(&value))
+        }
```

Comment:
Just as an example, replacing `Number` with `f64` reduces this to
```rust
Static(components) => Value::Fixed(f(&components)),
```

Reply:
Great suggestion, applied.